cloudinit/config/schema.py

```diff
@@ -1119,11 +1119,13 @@ def validate_cloudconfig_file(
         if not cloudconfig.get("network", cloudconfig):
             print("Skipping network-config schema validation on empty config.")
             return False
-        if netplan_validate_network_schema(
-            network_config=cloudconfig, strict=True, annotate=annotate
-        ):
-            return True  # schema validation performed by netplan
-        if network_schema_version(cloudconfig) != 1:
+        network_version = network_schema_version(cloudconfig)
+        if network_version == 2:
+            if netplan_validate_network_schema(
+                network_config=cloudconfig, strict=True, annotate=annotate
+            ):
+                return True  # schema validation performed by netplan
+        if network_version != 1:
```

Comment:
Nit: Do we need this `if` statement? Everything under it will be under the `if network_version == 2` block, which I think would be adequate?

Reply:
On systems without netplan but with network-config version 2, the `netplan_validate_network_schema` will have returned False on line 1123. In those cases, we want to fail with an informative message like `No network schema for version 2`. Instead of a json schema validation error that isn't as informative `Invalid network schema: 'version' not in [1]`. Also, in the case we are version 1 here, we want to fall through to the standard `validate_cloudconfig_schema` below to actually validate that network-config v1 file instead of claiming to skip it and early return

Reply:
Yeah, I think I understand that. I'm just saying we have this now:
```python
        if network_version == 2:
            if netplan_validate_network_schema(
                network_config=cloudconfig, strict=True, annotate=annotate
            ):
                return True  # schema validation performed by netplan
        if network_version != 1:
            # Validation requires JSON schema definition in
            # cloudinit/config/schemas/schema-network-config-v1.json
            print(
                "Skipping network-config schema validation."
                " No network schema for version:"
                f" {network_schema_version(cloudconfig)}"
            )
            return False
```

Isn't `if network_version == 2` effectively the same as `if network_version != 1`? Couldn't it be one block?
```python
        if network_version == 2:
            if netplan_validate_network_schema(
                network_config=cloudconfig, strict=True, annotate=annotate
            ):
                return True  # schema validation performed by netplan
            # Validation requires JSON schema definition in
            # cloudinit/config/schemas/schema-network-config-v1.json
            print(
                "Skipping network-config schema validation."
                " No network schema for version:"
                f" {network_schema_version(cloudconfig)}"
            )
            return False
```